common/lib/client/realtimechannel.js

```diff
@@ -28,6 +28,9 @@ var RealtimeChannel = (function() {
 			 * I.e. all encodings added by Ably Realtime have been decoded. */
 			baseEncodedPreviousPayload: undefined
 		};
+		this.lastPayloadMessageId = null;
```

Comment:
Out of interest @SimonWoolf, under what conditions do we prefix attributes with `_`?  
@codemerx could we not reduce the attribute count on the channel object by doing something like:

```javascript
this.lastPayload = {
  messageId: null,
  protocolMessageChannelSerial: null, 
  decodeFailureRecoveryInProgress: false
}
```

Reply:
> Out of interest @SimonWoolf, under what conditions do we prefix attributes with _?

Nominally private methods/attributes on public objects should be prefixed with underscores, see https://github.com/ably/ably-js/issues/371 . But until we have a major version release we're only doing that for new ones added as would be a breaking change if anyone was using them.

Reply:
OK, so then we should be doing this for all new private methods right, and these additions fall into that category @codemerx.

Reply:
Refactored in https://github.com/ably/ably-js/pull/615/commits/f514708ebc10edf3c9faa748bc0506b745a94caf